PreMailer.Net/PreMailer.Net/PreMailer.cs

```diff
@@ -405,7 +417,7 @@ private void RemoveStyleElements(IEnumerable<IElement> cssSourceNodes, bool pres
 				Selector = selectorParser.ParseSelector(x.Value.Name)
 			}).Where(x => x.Selector != null).ToList();
 
-			foreach (var el in _document.DescendentsAndSelf<IElement>())
+			foreach (var el in _document.Body.DescendentsAndSelf<IElement>())
```

Comment:
I'm unsure if we create problems by not allowing to target the `html` tag? I cannot rule out the use of obscure hacks on the `html` element for some email clients is necessary?

Would need some more research to be honest. You could also avoid the style tags on those elements, with `body *`?